core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -45,6 +45,33 @@
 
 public class TableMetadataParser {
 
+  public enum Codec {
```

Comment:
Can this be private or package-private?

Reply:
I've started with private but I am iterating through codecs in `HadoopTableOperations`. Any thoughts?

Reply:
That's fine.